src/k-provider/common/data-loader-manager.js

```diff
@@ -64,19 +75,17 @@ export default class DataLoaderManager {
       this._multiRequest.execute().then(
         response => {
           this._multiResponse = response;
-          if (!response.success) {
-            reject(response);
+          let preparedData: Object = this.prepareData(response);
+          if (preparedData.success) {
+            resolve(this._loaders);
           } else {
-            let preparedData: Object = this.prepareData(response);
-            if (preparedData.success) {
-              resolve(this._loaders);
-            } else {
-              reject({success: false, data: preparedData.error});
-            }
+            reject(
+              new Error(Error.Severity.CRITICAL, Error.Category.NETWORK, Error.Code.BAD_SERVER_RESPONSE, {success: false, data: preparedData.error})
```

Comment:
this error should be named `API_RESPONSE_MISMATCH` and should contain the headers data